src/scripts/eastern_kingdoms/elwynn_forest/elwynn_forest.cpp

```diff
@@ -14,80 +14,149 @@
+
         m_creature->SetUInt32Value(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_DEAD);
-        m_creature->SetStandState(UNIT_STAND_STATE_DEAD);   // lay down
-        spellHit = false;
+        m_creature->SetStandState(UNIT_STAND_STATE_DEAD);
+        m_creature->SetDeathState(JUST_DIED);
     }
 
-    void MoveInLineOfSight(Unit *who) override { }
-
-    void UpdateAI(uint32 const diff) override
+    void UpdateAI(const uint32 uiDiff) override
     {
         if (m_creature->IsStandingUp())
         {
-            if (lifeTimer < diff)
-                m_creature->AI()->EnterEvadeMode();
-            else
-                lifeTimer -= diff;
+            switch (m_uiState)
+            {
+            case tome_of_divinity::STATE_JUST_REVIVED:
+            {
+                if (m_uiTalkTimer < uiDiff)
+                {
+                    if (Player * pPlayer{ sObjectAccessor.FindPlayer(m_playerGuid) })
+                    {
+                        m_creature->SetFacingToObject(pPlayer);
+                        DoScriptText(tome_of_divinity::SAY_HEAL, m_creature, pPlayer);
+                    }
+
+                    m_playerGuid.Clear();
+
+                    m_uiState = tome_of_divinity::STATE_DO_EMOTE;
+                }
+                else
+                {
+                    m_uiTalkTimer -= uiDiff;
+                }
+
+                break;
+            }
+            case tome_of_divinity::STATE_DO_EMOTE:
+            {
+                if (m_uiEmoteTimer < uiDiff)
+                {
+                    m_creature->HandleEmote(EMOTE_ONESHOT_BOW);
+                    m_uiState = tome_of_divinity::STATE_RESET;
+                }
+                else
+                {
+                    m_uiEmoteTimer -= uiDiff;
+                }
+
+                break;
+            }
+            case tome_of_divinity::STATE_RESET:
+                [[fallthrough]];
+            default:
+            {
+                if (m_uiResetTimer < uiDiff)
+
+                {
+                    tome_of_divinityAI::Reset();
+                }
+                else
+                {
+                    m_uiResetTimer -= uiDiff;
+                }
+
+                break;
+            }
+            }
         }
     }
 
     void SpellHit(SpellCaster* pCaster, SpellEntry const* pSpellEntry) override
     {
-        if (pSpellEntry->Id == 8593 && !spellHit)
+        if (pSpellEntry->Id == tome_of_divinity::SPELL_SYMBOL_OF_LIFE && !m_bSpellHit)
         {
             m_creature->SetStandState(UNIT_STAND_STATE_STAND);
             m_creature->SetUInt32Value(UNIT_DYNAMIC_FLAGS, 0);
-            //m_creature->RemoveAllAuras();
-            DoScriptText(SAY_HEAL, m_creature, pCaster->ToUnit());
-            spellHit = true;
+
+            m_uiState = tome_of_divinity::STATE_JUST_REVIVED;
+
+            if (Player * pPlayer{ pCaster->ToPlayer() })
+            {
+                m_playerGuid = static_cast<ObjectGuid>(pPlayer->GetGUID());
```

Comment:
Why are you calling GetGUID which returns the guid as raw uint64 and then casting it to ObjectGuid instead of just using GetObjectGuid which does what you want?